### PR TITLE
feat(ingestion-ui) Add LookML form to managed ingestion UI

### PR DESCRIPTION
**Pull request by chriscollins3456**
Adds the form for LookML and a new field type to accommodate this form - the Dict type. This is where a field's value is another dictionary. This dictionary is extra special because it's really a dictionary of dictionaries where one field is the key for the children dictionaries. `connection_to_platform_map` is a dictionary where the `Name` field is the key of the children dictionaries (check out screenshot below).

This also adds validation on fields to ensure that everything that needs to be filled out is (including complex validation saying you need one subset of fields or another)

**The new form**
image

**When adding a connection to platform map:**
image

**The yaml as a result of filling out `connection_to_platform_map` in the above:**
image

**multiple maps filled out:**
image

**When the whole form is filled out:**
image

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/constants.ts [hunk 1]
```diff
@@ -78,6 +78,23 @@ import { POSTGRES } from '../../conf/postgres/postgres';
...
+    LOOKML_CLIENT_ID,
+    LOOKML_CLIENT_SECRET,
+    LOOKML_GITHUB_INFO_REPO,
+    PARSE_TABLE_NAMES_FROM_SQL,
+    PROJECT_NAME,
+} from './lookml';
+
+export enum RecipeSections {
+    Connection = 0,
+    Filter = 1,
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/ingest/source/builder/RecipeForm/constants.ts:95]: Is filter being used anywhere? What is it for?

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/FormField.tsx [hunk 2]
```diff
@@ -43,8 +31,8 @@ interface SelectFieldProps {
 
 function ListField({ field, removeMargin }: ListFieldProps) {
     return (
-        <Form.List name={field.name}>
-            {(fields, { add, remove }) => (
+        <Form.List name={field.name} rules={field.rules || undefined}>
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/ingest/source/builder/RecipeForm/FormField.tsx:34]: I love that we're adding rule-based validation here

**Review by aditya-radhakrishnan — APPROVED** [review]
LGTM overall!

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/constants.ts (hunk 1 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/ingest/source/builder/RecipeForm/constants.ts:95]: yup! most forms have the filter section - it's where you add allow/deny patters

File: datahub-web-react/src/app/ingest/source/builder/sources.json [hunk 3]
```diff
@@ -34,6 +34,13 @@
         "docsUrl": "https://datahubproject.io/docs/generated/ingestion/sources/looker/",
         "recipe": "source:\n    type: looker\n    config:\n        # Coordinates\n        base_url: # Your Looker instance URL, e.g. https://company.looker.com:19999\n\n        # Credentials\n        # Add secret in Secrets Tab with relevant names for each variable\n        client_id: \"${LOOKER_CLIENT_ID}\" # Your Looker client id, e.g. admin\n        client_secret: \"${LOOKER_CLIENT_SECRET}\" # Your Looker password, e.g. password_01"
     },
+    {
+        "urn": "urn:li:dataPlatform:lookml",
+        "name": "lookml",
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/ingest/source/builder/sources.json:39]: Nice so simple!

File: datahub-web-react/src/app/ingest/source/builder/sources.json [hunk 4]
```diff
@@ -34,6 +34,13 @@
         "docsUrl": "https://datahubproject.io/docs/generated/ingestion/sources/looker/",
         "recipe": "source:\n    type: looker\n    config:\n        # Coordinates\n        base_url: # Your Looker instance URL, e.g. https://company.looker.com:19999\n\n        # Credentials\n        # Add secret in Secrets Tab with relevant names for each variable\n        client_id: \"${LOOKER_CLIENT_ID}\" # Your Looker client id, e.g. admin\n        client_secret: \"${LOOKER_CLIENT_SECRET}\" # Your Looker password, e.g. password_01"
     },
+    {
+        "urn": "urn:li:dataPlatform:lookml",
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/ingest/source/builder/sources.json:38]: How do we find the logo forthis guy?

**Review by jjoyce0510 — APPROVED** [review]
LGTM ! Nice work.

File: datahub-web-react/src/app/ingest/source/builder/sources.json (hunk 4 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/ingest/source/builder/sources.json:38]: answered off of github - I have a mapping for this to looker which i have a mapping to get the logo from there

**Pull request merged**